packages/foam-workspace-manager/src/utils/utils.ts

```diff
@@ -1,23 +1,103 @@
-// @todo convert this to use ast parsing
-
-// import unified from 'unified';
-// import markdown from 'remark-parse';
-// import wikiLinkPlugin from 'remark-wiki-link';
-// let processor = unified()
-//   .use(markdown, { gfm: true })
-//   .use(wikiLinkPlugin);
-
-import { findTopLevelHeading, findWikilinksInMarkdown } from './markdown-utils';
+import * as path from 'path';
+import unified from 'unified';
+import markdownParse from 'remark-parse';
+import wikiLinkPlugin from 'remark-wiki-link';
+import visit, { CONTINUE, EXIT } from 'unist-util-visit';
+import { Node, Parent } from 'unist';
+import { BubbleLink, Bubble, Foam } from '../core';
 
 // @ts-expect-error
 export function readWorkspaceFile(filename: string): string {
   throw new Error('Not implemented');
 }
 
+// pipeline cache
+let processor: unified.Processor | null = null;
+
+function parse(markdown: string): Node {
+  processor =
```

Comment:
this code I copied from @jojanaho's PR - unfortunately because the PR is on a fork I couldn't (or don't know how) create this branch off of it